extensions/resource-deployment/src/ui/modelViewUtils.ts

```diff
@@ -503,79 +523,33 @@ function processDropdownOptionsTypeField(context: FieldContext): azdata.DropDown
 		label: context.fieldInfo.label
 	});
 	dropdown.fireOnTextChange = true;
-	context.onNewInputComponentCreated(context.fieldInfo.variableName!, { component: dropdown });
+	context.onNewInputComponentCreated(context.fieldInfo.variableName || context.fieldInfo.label, { component: dropdown });
 	addLabelInputPairToContainer(context.view, context.components, label, dropdown, context.fieldInfo);
 	return dropdown;
 }
 
 function processDateTimeTextField(context: FieldContext): void {
-	const label = createLabel(context.view, { text: context.fieldInfo.label, description: context.fieldInfo.description, required: context.fieldInfo.required, width: context.fieldInfo.labelWidth, cssStyles: context.fieldInfo.labelCSSStyles });
-	const defaultValue = context.fieldInfo.defaultValue + getDateTimeString();
-	const input = context.view.modelBuilder.inputBox().withProperties<azdata.InputBoxProperties>({
-		value: defaultValue,
-		ariaLabel: context.fieldInfo.label,
-		inputType: 'text',
-		required: context.fieldInfo.required,
-		placeHolder: context.fieldInfo.placeHolder
-	}).component();
-	input.width = context.fieldInfo.inputWidth;
-	context.onNewInputComponentCreated(context.fieldInfo.variableName!, { component: input });
-	addLabelInputPairToContainer(context.view, context.components, label, input, context.fieldInfo);
+	context.fieldInfo.defaultValue = context.fieldInfo.defaultValue + getDateTimeString();
+	const input = createInputBoxField({ context });
+	context.onNewInputComponentCreated(context.fieldInfo.variableName || context.fieldInfo.label, { component: input });
 }
 
 function processNumberField(context: FieldContext): void {
-	const label = createLabel(context.view, { text: context.fieldInfo.label, description: context.fieldInfo.description, required: context.fieldInfo.required, width: context.fieldInfo.labelWidth, cssStyles: context.fieldInfo.labelCSSStyles });
-	const input = createNumberInput(context.view, {
-		defaultValue: context.fieldInfo.defaultValue,
-		ariaLabel: context.fieldInfo.label,
-		min: context.fieldInfo.min,
-		max: context.fieldInfo.max,
-		required: context.fieldInfo.required,
-		width: context.fieldInfo.inputWidth,
-		placeHolder: context.fieldInfo.placeHolder
+	const input = createInputBoxField({ context, inputBoxType: 'number' });
+	context.onNewInputComponentCreated(context.fieldInfo.variableName || context.fieldInfo.label, {
+		component: input,
+		inputValueTransformer: (value: string | number | undefined) => (typeof value === 'string') && value.length > 0 ? parseFloat(value) : value
```

Comment:
We need to make sure to do numeric comparisons numerically instead of string sort order comparison, so cast output of numeric field to number before the value gets used downstream